### ISSUE TITLE
ESP32 thinks wifi is not connected, fails to start

### ISSUE THREAD

**Issue by ingenium13** [DanielOgorchock/ST_Anything#115]
I just flashed st_anything to my esp32 (using the sample st_anything_multiples_esp32wifi sketch), but it seems to not recognize that it's connected to wifi. The following shows in my serial monitor, with debug level set to verbose:
```
entry 0x40078a58
Everything: init started
Everything: Free RAM = -1
Disabling ESP32 WiFi Access Point

[D][WiFiGeneric.cpp:293] _eventCallback(): Event: 2 - STA_START
[WiFi-event] event: 2
ESP32 station start

Initializing ESP32 WiFi network.  Please be patient...
Attempting to connect to WPA SSID: MUSHROOM-2.4
.......[D][WiFiGeneric.cpp:293] _eventCallback(): Event: 7 - STA_GOT_IP
[WiFi-event] event: 7
WiFi connected
IP address: 
192.168.42.34
[D][WiFiGeneric.cpp:293] _eventCallback(): Event: 7 - STA_GOT_IP
[WiFi-event] event: 7
WiFi connected
IP address: 
192.168.42.34
....what is taking so long?
...........what is taking so long?
...........what is taking so long?
...........what is taking so long?
...........what is taking so long?
...........what is taking so long?
...........what is taking so long?
```
I can ping the esp32's IP from other devices on my network no problem. Looking at the code in SmartThingsESP32WiFi.cpp on line 135, it seems that this is caused by "WiFi.status() != WL_CONNECTED" always being true so it never exits the loop. However, wifi is indeed connected, and the device is accessible over the network.

**Comment by vseven** [comment]
Yeah...there is some weirdness with the ESP32 and WiFi that I have been going back and forth about for 6 months now with the developers of the chip.  There has to be a lot of hoops to jump through to keep it connected.  Some are implemented in the ST_Anything version of the libraries and some still need to be but its a little confusing as to what.  See Issue 102 which I already posted.

For me I set a reservation on my router in DHCP so the ESP32 always gets the same IP address which works in place of hard coding it in the device/sketch.
